### DISCUSSION TITLE
How to download job logs programmatically from  developer.mend.io ?

### DISCUSSION THREAD

**Discussion by gberche-orange** [renovatebot/renovate#24161] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'd like to automate the downloading of job logs in order to automatically detect config regressions, e.g. with a lot of skipped updated such as  `"skipReason": "unknown-registry"`

However, I did not yet find a documented API to fetch the job logs.

### What you tried so far.

I copy from the browser the download Url as curl and replaying into a terminal to save the resulting json output.

![image](https://github.com/renovatebot/renovate/assets/4748380/eaa0e132-c77e-449b-ab29-ced04629cde7)

 The `If-None-Match` header needs to be manually removed from the curl command. Then I can process the json output using jq see https://github.com/orange-cloudfoundry/paas-templates/issues/2000#issuecomment-1697820474 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
There is no API, and we don't want any programmatic scraping

**Reply by gberche-orange** [reply]
What's the suggested way for a team to detect regressions in their renovate.json config (e.g. as they rename/move files, breaking the include/exclude paths) other than through analyzing logs ?

Are you considering the dashboard issue to publish metrics similar to the following in the future ?

```
less renovate-logs.json | jq -r .logs | jq | grep skipReason | sort | uniq -c | sort -n
      1             "skipReason": "invalid-value"
      4             "skipReason": "invalid-name",
      5             "skipReason": "unversioned-reference",
     13             "skipReason": "invalid-version",
     18             "skipReason": "ignored"
     21             "skipReason": "unspecified-version",
     25             "skipReason": "unknown-registry",
```

**Reply by rarkins** [reply]
At this point I don't have any suggested way to detect regressions/human error apart from the teams themselves paying attention to the dependency dashboard issue

**Comment by Charlotte-br560** [comment]
Have you explored using [Scraper API](https://crawlbase.com/scraper-api-auto-parse-web-data) for fetching job logs programmatically? It might offer a more reliable and scalable solution compared to manual URL extraction.
